injectproxy/routes.go

```diff
@@ -239,6 +256,26 @@ func (r *routes) enforceLabel(h http.HandlerFunc) http.Handler {
 	})
 }
 
+// getLabelValue returns the statically set label value, or the label value
+// sent through a URL parameter.
+// It returns an error when either the value is found in both places, or is not found at all.
+func (r *routes) getLabelValue(req *http.Request) (string, error) {
+	formValue := req.FormValue(r.label)
+	if r.labelValue != "" && formValue != "" {
+		return "", fmt.Errorf("a static value for the %s label has already been specified", r.label)
+	}
+
+	if r.labelValue == "" && formValue == "" {
+		return "", fmt.Errorf("bad request. The %q query parameter must be provided", r.label)
```

Comment:
I would simplify the error message
```suggestion
		return "", fmt.Errorf("the %q query parameter must be provided", r.label)
```